feat: add settings window with word/character count toggle

smolyaninov opened the pull request:
Settings window added

You can now enable or disable the word/character counter.
The setting is saved and applied on the next launch.

In the future, the window should be made draggable, and the minimize/maximize buttons should be removed.

Thanks for the opportunity to explore and try out new tools — it’s been an interesting experience! 🤓🧪

smolyaninov committed:
feat: add settings window with word/character count toggle

smolyaninov committed:
feat: add settings window with word/character count toggle

arikchakma commented:
Instead of a separate settings page, I was planning to add `cmd+k` overlay where you can toggle all of these settings. Because I wanna remove mouse navigation as much as possible. Let me know what do you think.

smolyaninov committed:
refactor: remove settings window

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov commented:
Absolutely, your approach makes more sense and feels more natural for this kind of interface.
I’ve removed the separate settings window and moved the toggles into the cmd+k overlay instead.

Thanks for pointing it out!

<img width="512" height="346" alt="cmdk" src="https://github.com/user-attachments/assets/63b89055-1b60-4cff-9902-3cface708b72" />

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

arikchakma commented:
I will look into this hopefully by this weekend😊

arikchakma commented:
@smolyaninov why not implement it similar to the select note dialog? It should also auto resize based on the minimum size.

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov committed:
feat: add cmd+K command palette

smolyaninov commented:
@arikchakma I’ve now implemented it in a similar way 😊

arikchakma commented:
Finished my exam man, I will have a look into it next weekend😊

arikchakma commented:
I will refactor it, I think there are some scope to improve some functionalities. Thank you for the PR.

arikchakma committed:
chore: remove comment
